build-support/githooks/pre-commit

```diff
@@ -78,7 +75,7 @@ echo "* Checking packages"
 ./build-support/bin/check_packages.sh "${DIRS_TO_CHECK[@]}" || exit 1
 
 echo "* Checking headers"
-./build-support/bin/check_header.py "${DIRS_TO_CHECK[@]}" --files-added "${ADDED_FILES[@]}" || exit 1
+./pants --changed-since="${MERGE_BASE}" validate || exit 1
```

Comment:
NB that this will only check files belonging to targets. A neat feature of `validate` is being able to check files with no owning targets, like BUILD flies. It's fine to not do in this PR, but it would be nice for us to validate BUILD files.